faq/manage-cluster-faq.md

```diff
@@ -35,16 +37,20 @@ By default, TiDB/PD/TiKV outputs standard error in the logs. If a log file is sp
 
 ### How to safely stop TiDB?
 
-Kill all the services using `kill` directly. The components of TiDB will do `graceful shutdown`.
+- A load balancer is running (recommended): Stop the load balancer and then run the kill command. Then TiDB waits for a period specified by [graceful-wait-before-shutdown](/tidb-configuration-file.md#graceful-wait-before-shutdown-new-in-v50) until all sessions are terminated. Then TiDB stops running.
+
+- No load balancer is running: Run the kill command. The TiDB components are gracefully stopped.
 
 ### Can `kill` be executed in TiDB?
 
-- You can `kill` DML statements. First use `show processlist` to find the ID corresponding with the session, and then run `kill tidb [session id]`.
-- You can `kill` DDL statements. First use `admin show ddl jobs` to find the ID of the DDL job you need to kill, and then run `admin cancel ddl jobs 'job_id' [, 'job_id'] ...`. For more details, see the [`ADMIN` statement](/sql-statements/sql-statement-admin.md).
+[Kill](/sql-statements/sql-statement-kill.md) commands are disabled by default. To run a kill command, you need to set `compatible-kill-query = true` or enable Global Kill (experimental) first. Then proceed with the following steps:
+
+- Kill DML statements: First use `show processlist` to find the ID corresponding with the session, and then run `kill tidb [session id]`.
+- Kill DDL statements: First use `admin show ddl jobs` to find the ID of the DDL job you need to kill, and then run `admin cancel ddl jobs 'job_id' [, 'job_id'] ...`. For more details, see the [`ADMIN` statement](/sql-statements/sql-statement-admin.md).
```

Comment:
This advice is not technically correct. It says that you need to enable `compatible-kill-query` or global kill *to then use* `kill tidb`. But rather, `kill tidb` works if you don't enable either of these features. It is `kill` (without the tidb) that requires these to be enabled.

Reply:
@easonn7 PTAL, thanks.

Reply:
@morgo This modification is according to your comment in https://github.com/pingcap/docs/issues/6536:

The answer is not really helpful. KILL can be used, but is disabled by default because of the case that you could be using a load balancer and killing the wrong connection. This can be changed by either enabling compatible-kill-query = true, or enabling global kill (experimental).

Reply:
@xuyifangreeneyes Could you please take a look at this FAQ? Thanks.

Reply:
1. `kill tidb` always works. However, if `enable-global-kill = false`, `kill tidb` works only when the kill command sends to the right TiDB instance and may cause wrong kill when there is a proxy between the client and TiDB instances. Hence we don't recommend users to use `kill tidb` unless `enable-global-kill = true` or they know they send the kill command to the right TiDB instance.
2. We need to address that we don't recommend users to set `compatible-kill-query = true`.
3. Global kill will be GA in v6.1.0 and there will be a new config `enable-global-kill` whose default value is true(I'm just writing the pr...). In some previous versions, the switch that controls global kill is an invisible config `experimental.enable-global-kill`, which is also not mentioned in doc. Even if we set `experimental.enable-global-kill = true`, global kill may not work due to the existing bug(https://github.com/pingcap/tidb/issues/33538)

Reply:
4. The fix pr of #33538 will be cherry picked to the previous versions.

Reply:
It would help if the FAQ described why KILL is disabled by default (load balancer could route to wrong server), and when it is safe to enable compatible kill (only when there is no load balancer). Some users might require a compatible kill because they have a program to kill idle connections (using something like pt-kill is common with mysql).

The current wording is not helpful, because it says you need to enable various settings (without explaining) but it then uses the example syntax which doesn't require these settings (kill tidb).

Reply:
> It would help if the FAQ described why KILL is disabled by default (load balancer could route to wrong server), and when it is safe to enable compatible kill (only when there is no load balancer). Some users might require a compatible kill because they have a program to kill idle connections (using something like pt-kill is common with mysql).
> 
> The current wording is not helpful, because it says you need to enable various settings (without explaining) but it then uses the example syntax which doesn't require these settings (kill tidb).

Got it. We will optimize the wording and provide a clearer version.

Reply:
@morgo I have updated the answers to [How to safely stop TiDB](https://github.com/pingcap/docs/pull/8638/files#diff-99d1372119ae4ed697e8d7fda222067f376d7bc2881ddce7d4bec353c2d00b05R38) and [Can `Kill` be executed in TiDB](https://github.com/pingcap/docs/pull/8638/files#diff-99d1372119ae4ed697e8d7fda222067f376d7bc2881ddce7d4bec353c2d00b05R44). PTAL. Thanks.

Reply:
LGTM